cli/tests/functional/src/test/java/io/helidon/build/cli/tests/CliDistributionTest.java

```diff
@@ -0,0 +1,132 @@
+ * limitations under the License.
+ */
+package io.helidon.build.cli.tests;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import io.helidon.build.common.InputStreams;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+import static io.helidon.build.cli.tests.FunctionalUtils.getProperty;
+import static io.helidon.build.cli.tests.FunctionalUtils.setMavenLocalRepoUrl;
+import static io.helidon.build.common.FileUtils.list;
+import static io.helidon.build.common.FileUtils.unzip;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * CLI distribution tests.
+ */
+public class CliDistributionTest {
+
+    private static Path distDir;
+    private static final String CLI_VERSION_KEY = "cli.version";
+
+    @BeforeAll
+    static void setup() throws IOException {
+        setMavenLocalRepoUrl();
+        distDir = Files.createTempDirectory("dist");
+        Path targetDir = Path.of(getProperty("helidon.executable.directory"));
+        Path cliZip = targetDir.resolve("target/distribution/helidon.zip");
+        unzip(cliZip, distDir);
+        list(distDir, 3).stream()
+                .map(Path::toFile)
+                .forEach(file -> file.setExecutable(true));
+    }
+
+    @Test
+    void testCliContent() {
+        List<String> content = list(distDir, 4).stream()
+                .peek(p -> assertThat(Files.exists(p), is(true)))
+                .map(p -> distDir.relativize(p))
+                .map(Path::toString)
+                .map(s -> s.replace("\\", "/"))
+                .collect(Collectors.toList());
+
+        //Ensure main directory are present
+        assertThat(content, hasItems("helidon/bin"));
+        assertThat(content, hasItems("helidon/lib"));
+        assertThat(content, hasItems("helidon/lib/libs"));
+        //Ensure main files are present
+        assertThat(content, hasItems("helidon/bin/helidon"));
+        assertThat(content, hasItems("helidon/bin/helidon.bat"));
+        assertThat(content, hasItems("helidon/lib/helidon.jar"));
+        assertThat(content, hasItems("helidon/LICENSE.txt"));
+    }
+
+    @Test
+    @EnabledOnOs(value={OS.LINUX, OS.MAC}, disabledReason = "Run only on Mac or Linux")
+    void testShellCreateProject() throws IOException {
+        runCreateProjectTest(distDir.resolve("helidon/bin/helidon").toString());
+    }
+
+    @Test
+    @EnabledOnOs(value={OS.WINDOWS}, disabledReason = "Run only on Windows")
+    void testBatchCreateProject() throws IOException {
+        runCreateProjectTest(distDir.resolve("helidon\\bin\\helidon.bat").toString());
```

Comment:
IIRC you don't have to use backslash, the path will be translated by the windows `FileSystem`.